synapse/handlers/register.py

```diff
@@ -406,6 +412,32 @@ def register_email(self, threepidCreds):
                     403, "Third party identifier is not allowed"
                 )
 
+    @defer.inlineCallbacks
+    def bind_email_to_account(self, user_id, email, validated_at=None):
+        """Binds an email address with the registered identity service.
+
+        Args:
+            user_id (str): The user id of the user to bind the email to.
+            email (str): The email to bind.
+            validated_at (int|None): Time to mark emails validated at. If
+                `None`, the current time is used.
+
+        Returns:
+            Deferred
+        """
+
+        # generate threepid dict
+        threepid_dict = {
+            "medium": "email",
+            "address": email,
+            "validated_at": self.hs.get_clock().time_msec(),
+        }
+
+        # Bind email to new account
+        yield self._register_email_threepid(
```

Comment:
for the record: where you have a function which just does some synchronous things and ends up calling an asynchronous function, you can skip `@inlineCallbacks` and just return the deferred from the other function:

```
def bind_email_to_account():
    do_synchronous_stuff()
    return self._register_email_threepid(...)
```

Reply:
Good to know, thanks!